go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/kubeflow/spark-operator
 
-go 1.22.5
+go 1.23.0
```

Comment:
Do we need the minor version or can we use simply 1.23?

Reply:
Definitely not strictly necessary given Go's backward compatibility guarantees and not pinning should reduce the frequency of toolchain upgrade PRs. Updated

Reply:
I think the patch version is needed, otherwise `go mod tidy` will fail:

```bash
$ go mod tidy
go: downloading go1.23 (darwin/arm64)
go: download go1.23 for darwin/arm64: toolchain not available
```

There are some issues related to the toolchain version not being available: https://github.com/golang/go/issues/62278.